### PR TITLE
Show exra information in validation error messages

### PR DESCRIPTION
**Pull request by tarmolov**
/cc @dfilatov @zloylos

### CODE REVIEW

File: lib/validators/normalize.js [hunk 1]
```diff
@@ -39,7 +40,11 @@ module.exports = function (value, name, declaration) {
         case 'number':
             var parsedValue = parseFloat(value);
             if (Number.isNaN(parsedValue)) {
-                throw new ApiError(ApiError.BAD_REQUEST, 'invalid value for ' + name + ' parameter');
+                var errorMessage = util.format(
+                    'invalid value for "%s" parameter, expected type "%s" but got "%d"',
```
Comment (zloylos) [inline lib/validators/normalize.js]: why do you using `%d` if value was be NaN?

Comment (tarmolov) [inline lib/validators/normalize.js]: Do you suggest using `%s`?

%d -> NaN
%s -> nan

Comment (zloylos) [inline lib/validators/normalize.js]: You used `value`, not `parsedValue`,
`%s for value variable may be string or Object or undefined etc`
`%s for parsedValue always be NaN`

**Comment by dfilatov** [comment]
lgtm except copy-paste

File: lib/validators/normalize.js [hunk 2]
```diff
@@ -39,7 +40,11 @@ module.exports = function (value, name, declaration) {
         case 'number':
             var parsedValue = parseFloat(value);
             if (Number.isNaN(parsedValue)) {
-                throw new ApiError(ApiError.BAD_REQUEST, 'invalid value for ' + name + ' parameter');
+                var errorMessage = util.format(
+                    'invalid value for "%s" parameter, expected type "%s" but got "NaN"',
```
Comment (zloylos) [inline lib/validators/normalize.js]: Let's show got type: `typeof value`

Comment (dfilatov) [inline lib/validators/normalize.js]: but please keep value as well

**Pull request merged**